modules/rank-eval/src/internalClusterTest/java/org/elasticsearch/index/rankeval/RankEvalRequestIT.java

```diff
@@ -291,7 +291,7 @@ public void testIndicesOptions() {
 
         // test expand_wildcards
         request = new RankEvalRequest(task, new String[] { "tes*" });
-        request.indicesOptions(IndicesOptions.fromParameters("none", null, null, "false", SearchRequest.DEFAULT_INDICES_OPTIONS));
+        request.indicesOptions(IndicesOptions.fromParameters("none", "true", null, "false", SearchRequest.DEFAULT_INDICES_OPTIONS));
```

Comment:
The ITs and BWC tests that are failing (eg https://gradle-enterprise.elastic.co/s/ahtc43n4422im) are all about this case:
given a **single** wildcard, and the following request options `expand_wildcards=none`, and `ignore_unavailable=false` and `allow_no_indices=true`, the previous behavior would've resolved to the empty list (`allow_no_indices` is checked, and `ignore_unavailable` is ignored), but the new behavior throws an "index not found" exception (`ignore_unavailable` is checked, and `allow_no_indices` is ignored).

The issue is that an unexpanded wildcard (because `expand_wildcards=none`), when it is part of a larger expression is ignored if `ignore_unavailable=true` or throws a "not found" exception if `ignore_unavailable=false`, but if the expression only contains that single wildcard, the `ignore_unavailable` option is ignored and the wildcard is always ignored, and then, because the expression only contained that, is going to evaluate to empty, and then the `allow_no_indices` option is considered. I think this is a bug.

Reply:
We chatted in person and came to the same conclusion that this is a bug. Further, it is very rare case since `expand_wildcards=none` for a single item with a wildcard expression is valid but doesn't make much sense in practice.